src/jdk.crypto.mscapi/windows/classes/sun/security/mscapi/CPublicKey.java

```diff
@@ -93,7 +93,10 @@ public byte[] getEncoded() {
                     // ignore
                 }
             }
-            return encoding;
+
+            byte[] safeEncoding = new byte[encoding.length];
+            System.arraycopy(encoding, 0, safeEncoding, 0, encoding.length);
+            return safeEncoding;
```

Comment:
You can replace these lines (and below) with `return encoding.clone();` which is more compact and the same AFAICT.

Reply:
Good call, just pushed those changes.